### PR TITLE
New lint: needless_doc_main

### PR DESCRIPTION
**Pull request by llogiq**
changelog: Add `needless_doc_main` lint

### CODE REVIEW

File: clippy_lints/src/doc.rs [hunk 1]
```diff
@@ -68,6 +68,25 @@ declare_clippy_lint! {
...
+declare_clippy_lint! {
+    /// **What it does:** Checks for `fn main() { .. }` in doctests
+    ///
+    /// **Why is this bad?** The test can be shorter (and likely more readable)
+    /// if the `fn main()` is left implicit.
+    ///
+    /// **Known problems:** None.
+    ///
+    /// **Examples:**
+    /// /// ```rust
```
Comment (flip1995) [inline clippy_lints/src/doc.rs]:
```suggestion
    /// ```rust
```
Same below

Comment (llogiq) [reply clippy_lints/src/doc.rs]: On the contrary, this should be an example of a doctest within the docs.

Comment (flip1995) [reply clippy_lints/src/doc.rs]: Docs all the way down :turtle:

**Comment by flip1995** [comment]
Thanks!

@bors r+

**Pull request merged**